module/batch-action/src/Infrastructure/Persistence/Repository/Factory/DbalBatchActionFactory.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\BatchAction\Infrastructure\Persistence\Repository\Factory;
+
+use Ergonode\BatchAction\Domain\Entity\BatchAction;
+use Ergonode\BatchAction\Domain\Entity\BatchActionId;
+use Ergonode\BatchAction\Domain\ValueObject\BatchActionType;
+use Ergonode\BatchAction\Domain\ValueObject\BatchActionAction;
+
+class DbalBatchActionFactory
+{
+    /**
+     * @throws \ReflectionException
+     */
+    public function create(array $record): BatchAction
+    {
+        $reflector = new \ReflectionClass(BatchAction::class);
```

Comment:
I see no reason for not using the constructor.

Reply:
This notation allows for safe reconstruction of the object's structure without calling any methods, including the constructor, which prevents possible errors if there be any logic in it.

Reply:
I don't think it's safe - it's i.e. not resistant to changing the name of the internal property(like `type` to `specialType`). So it has drawbacks - just different ones than using constructor but in my opinion, it's just harder to catch/find those. I think unless there's a real justified reason for not using the public interface we should use it.